platform/ext/target/nordic_nrf/common/core/CMakeLists.txt

```diff
@@ -86,6 +86,7 @@ if (NRF_HW_INIT_RESET_ON_BOOT)
     target_compile_definitions(platform_s
         PUBLIC
             NRF_HW_INIT_RESET_ON_BOOT
+            $<$<BOOL:${NRF_HW_INIT_NRF_PERIPHERALS}>:NRF_HW_INIT_NRF_PERIPHERALS>
```

Comment:
not very clear what this does. but the rest looks fine

Reply:
It's a CMake generator expression if you are familiar with it?

Ref. https://gitlab.kitware.com/cmake/cmake/-/issues/18549#note_479462  for explanation why this is done.

Reply:
ahah. thanks for that.